pkg/armhelpers/disk.go

```diff
@@ -25,6 +23,7 @@ func (az *AzureClient) DeleteManagedDisk(ctx context.Context, resourceGroupName
 }
 
 // ListManagedDisksByResourceGroup lists managed disks in a resource group.
-func (az *AzureClient) ListManagedDisksByResourceGroup(ctx context.Context, resourceGroupName string) (result compute.DiskListPage, err error) {
```

Comment:
This is an example of a breaking change that will slow down delivery of this PR to master. If possible, implement what is needed while keeping the `ListManagedDisksByResourceGroup` method as-is.

Reply:
@jackfrancis 
This change can not be avoid.
DiskListPage belongs to package github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-10-01/compute, 
so I would need covert the return object from 2017-03-30/compute to 2018-10-01/compute

But the DiskListPage struct did not have public fields, I could not do the deep copy.

the current implementation is I add a wrapper to on top of DiskListPage and expose all the method it supports (for example Next())

// DiskListPage contains a page of Disk values.
type DiskListPage struct {
                fn func(context.Context, DiskList) (DiskList, error)
                dl DiskList
}